src/Serializer/AbstractItemNormalizer.php

```diff
@@ -96,6 +96,10 @@ public function hasCacheableSupportsMethod(): bool
      */
     public function normalize($object, $format = null, array $context = [])
     {
+        if (isset($context['resource_class']) && isset($context['normalize_resource_class'])) {
```

Comment:
Why not `input/output_class` anymore? It will be easier to understand, track and debug.